Update observation yamls for atm_jjob ens_run and var_run ctests

RussTreadon-NOAA opened the pull request:
`test_gdasapp_atm_jjob_var_run` and `test_gdasapp_atm_jjob_ens_run` fail in `develop` at 99f924e.   Two actions need to be taken to restore these tests to the `Passed` state:
1. update observation yamls used by these ctests to the new IODA format
2. update the observation dump files used by these ctests to the new IODA format

This PR addresses item 1.   The dump files used by these ctests reside in shared R2D2 databases on Hera and Orion.

Fixes #332

RussTreadon-NOAA committed:
update observation yamls for ens and var atm_jjob ctests (#332)

RussTreadon-NOAA commented:
The observation dump files used by the ens_run and var_run ctests

`R2D2_SHARED/ncdiag/ob/oper_gdas/PT6H/2021-03-23/ncdiag.oper_gdas.ob.PT6H.amsua_n19.2021-03-23T15:00:00Z.nc4`
`R2D2_SHARED/ncdiag/ob/oper_gdas/PT6H/2021-03-23/ncdiag.oper_gdas.ob.PT6H.sondes.2021-03-23T15:00:00Z.nc4`

have been updated to the new IODA format. 

The combination of the updated yamls in this PR along with the updated dump files result in `Passed` ctests as demonstrated from the Orion results below:
```
(gdasapp) Orion-login-4:/work/noaa/da/rtreadon/git/global_workflow/pr333/sorc/gdas.cd/build$ ctest -R test_gdasapp_atm_jjob
Test project /work/noaa/da/rtreadon/git/global_workflow/pr333/sorc/gdas.cd/build
    Start 1351: test_gdasapp_atm_jjob_var_prep
1/6 Test #1351: test_gdasapp_atm_jjob_var_prep ...   Passed    8.69 sec
    Start 1352: test_gdasapp_atm_jjob_var_run
2/6 Test #1352: test_gdasapp_atm_jjob_var_run ....   Passed  165.06 sec
    Start 1353: test_gdasapp_atm_jjob_var_post
3/6 Test #1353: test_gdasapp_atm_jjob_var_post ...   Passed    6.45 sec
    Start 1354: test_gdasapp_atm_jjob_ens_prep
4/6 Test #1354: test_gdasapp_atm_jjob_ens_prep ...   Passed    7.49 sec
    Start 1355: test_gdasapp_atm_jjob_ens_run
5/6 Test #1355: test_gdasapp_atm_jjob_ens_run ....   Passed  284.90 sec
    Start 1356: test_gdasapp_atm_jjob_ens_post
6/6 Test #1356: test_gdasapp_atm_jjob_ens_post ...   Passed    6.71 sec

100% tests passed, 0 tests failed out of 6

Total Test time (real) = 479.70 sec

```

Other ctests still fail
```
The following tests FAILED:
        1335 - test_gdasapp_soca_ana_run (Failed)
        1340 - test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_RUN (Failed)
        1342 - test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_VRFY (Failed)
        1346 - test_gdasapp_land_letkfoi_snowda (Failed)
```
These ctests are not within the scope of this PR.

ADCollard commented on parm/atm/obs/config/lgetkf_amsua_n19.yaml:
```diff
@@ -29,46 +29,46 @@ obs error:
     maxvalue: 27
   - variable:
-      name: brightness_temperature_1@ObsValue
+      name: brightnessTemperature_1@ObsValue
```
I thought we had got rid of the brightnessTemperature_<channum> format?

Also this should be ObsValue/brightnessTemperature, I think

ADCollard commented on parm/atm/obs/config/lgetkf_amsua_n19.yaml:
```diff
@@ -29,46 +29,46 @@ obs error:
   where:
   - variable:
-      name: scan_position@MetaData
+      name: sensorScanPosition@MetaData
```
Should be MetaData/sensorScanPosition

RussTreadon-NOAA replied:
Thanks @ADCollard .   I'll make this change and rerun the ctests.

CoryMartin-NOAA commented on parm/atm/obs/config/lgetkf_amsua_n19.yaml:
```diff
@@ -29,46 +29,46 @@ obs error:
   where:
   - variable:
-      name: scan_position@MetaData
+      name: sensorScanPosition@MetaData
```
Not required here, but just for @RussTreadon-NOAA 's awareness, this has been deprecated and should now be `MetaData/sensorScanPosition`

CoryMartin-NOAA replied:
Same applies to all of the `@` entries, they should be inverted and replaced with `/`

RussTreadon-NOAA replied:
Thanks @CoryMartin-NOAA for catching this.  My preference is to use the proper format.  I'll update the `@` entries as you indicate.

CoryMartin-NOAA reviewed (dismissed):
Approved, up to you @RussTreadon-NOAA if this should be merged or update the YAML per my comment.

RussTreadon-NOAA committed:
Replace @ notation with / (#332)

RussTreadon-NOAA commented:
Changes requested by @ADCollard and @CoryMartin-NOAA were committed at 5b3297e.

atm_jjob ctests rerun on Orion with all 6 tests passing
```
(gdasapp) Orion-login-4:/work/noaa/da/rtreadon/git/global_workflow/pr333/sorc/gdas.cd/build$ ctest -R test_gdasapp_atm_jjob
Test project /work/noaa/da/rtreadon/git/global_workflow/pr333/sorc/gdas.cd/build
    Start 1351: test_gdasapp_atm_jjob_var_prep
1/6 Test #1351: test_gdasapp_atm_jjob_var_prep ...   Passed   11.17 sec
    Start 1352: test_gdasapp_atm_jjob_var_run
2/6 Test #1352: test_gdasapp_atm_jjob_var_run ....   Passed  175.95 sec
    Start 1353: test_gdasapp_atm_jjob_var_post
3/6 Test #1353: test_gdasapp_atm_jjob_var_post ...   Passed    6.35 sec
    Start 1354: test_gdasapp_atm_jjob_ens_prep
4/6 Test #1354: test_gdasapp_atm_jjob_ens_prep ...   Passed    7.44 sec
    Start 1355: test_gdasapp_atm_jjob_ens_run
5/6 Test #1355: test_gdasapp_atm_jjob_ens_run ....   Passed  298.60 sec
    Start 1356: test_gdasapp_atm_jjob_ens_post
6/6 Test #1356: test_gdasapp_atm_jjob_ens_post ...   Passed    6.65 sec

100% tests passed, 0 tests failed out of 6

Total Test time (real) = 506.63 sec
```
One remaining question is whether or not to retain the `_{channel_number}` suffix still in `parm/atm/obs/config/lgetkf_amsua_n19.yaml`.   For example, is 
```
  - variable:
      name: ObsValue/brightnessTemperature_1
    minvalue: 50.0
    maxvalue: 550.0
```
how we want the set up our yamls moving forward?

CoryMartin-NOAA commented:
@RussTreadon-NOAA the changes to `amsua_n19` yaml and `lgetkf_amsua_n19` are not similar. I wonder if the `lgetkf` YAML is 'incorrect'. Are there any brightness_temperature_N variables in the `amsua_n19` YAML file? Presumably the QC should be identical between Var and EnKF? If so, perhaps just copy things from `amsua_n19` to `lgetkf_amsua_n19`?

ADCollard commented:
@RussTreadon-NOAA, I think if you replace all instances of filters with `brightnessTemperature_<channel_number>` with the single filter below, it will work (one reason we wanted to get rid of this syntax!)

```
- filter: Bounds Check
  filter variables:
   - name: brightnessTemperature
      channels: 1-15
  minvalue: 50.00001
  maxvalue: 449.99999
  action:
    name: reject
```

RussTreadon-NOAA commented:
> @RussTreadon-NOAA, I think if you replace all instances of filters with `brightnessTemperature_<channel_number>` with the single filter below, it will work (one reason we wanted to get rid of this syntax!)
> 
> ```
> - filter: Bounds Check
>   filter variables:
>    - name: brightnessTemperature
>       channels: 1-15
>   minvalue: 50.00001
>   maxvalue: 449.99999
>   action:
>     name: reject
> ```

This is concise and readable.  Nice change!   Thank you @ADCollard

guillaumevernieres reviewed (dismissed):
:tada:

RussTreadon-NOAA committed:
simplify lgetkf_amsua_n19.yaml for efficient use in test_gdasapp_atm_jjob_ens_run (#332)

RussTreadon-NOAA commented:
> @RussTreadon-NOAA the changes to `amsua_n19` yaml and `lgetkf_amsua_n19` are not similar. I wonder if the `lgetkf` YAML is 'incorrect'. Are there any brightness_temperature_N variables in the `amsua_n19` YAML file? Presumably the QC should be identical between Var and EnKF? If so, perhaps just copy things from `amsua_n19` to `lgetkf_amsua_n19`?

I agree.  var and enkf qc should be consistent.   I copied `amsua_n19` to `lgetkf_amsua_n19` and added `distribution` and `localizaton` sections.   `test_gdasapp_atm_jjob_ens_run` passed but took  748.10 seconds.  This is too long for a ctest.    

Seems we need two enkf yamls.   One has all the filters and checks we want for scientific cycling.  The other is for efficient ctests with only a few filters.   I removed all the filters found in `amsua_n19` from `lgetkf_amsua_n19` except for trimmed down versions of 
```
- filter: Background Check
- filter: Bounds Check
```
`test_gdasapp_atm_jjob_ens_run` ran faster but still took 426.67 sec.

I noticed that the original `lgetkf_amsua_n19.yaml` only simulated channels 4-6, 9-14.   I added this to revised `lgetkf_amsua_n19.yaml`.    This got the `test_gdasapp_atm_jjob_ens_run` run time down to 330.43 sec.    This is still high.

As one last change I reduced `fraction of retained variance` for the letkf from `0.975` to `0.750`.   This drops the run time down to 202.16 sec.    The change in the `fraction of retained variance` is in PR #311. 
  
The updated `lgetkf_amsua_n19.yaml` was committed at 697f3be.

RussTreadon-NOAA commented:
A check of observation dump files used on Hera by `test_gdasapp_atm_jjob_var_run ` and `test_gdasapp_atm_jjob_ens_run` shows variables to still be in the old IODA format.

For example, `gdas.t18z.amsua_n19.2021032318.nc4` contains `brightness_temperature`.   This should be `brightnessTemperature`.  

`sondes.2021032318.nc4` contains the winds as `eastward_wind` and `northward_wind`.

@CoryMartin-NOAA , we should check to rsync of R2D2_SHARED files from Orion to Hera.  It seems some files didn't make it to Hera.

RussTreadon-NOAA commented:
Conduct the following test on Hera

- install g-w with `feature/atm_jjob` for GDASApp.   Build with g-w ctests on.
- reproduce `test_gdasapp_atm_jjob_var_run ` and `test_gdasapp_atm_jjob_ens_run` failures
- rsync Orion `amsua_n19.2021032318.nc4` and `sondes.2021032318.nc4` to Hera ROTDIR
- rerun failed var and ens tests.   Both tests pass

```
(gdasapp) Hera(hfe04):/scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/pr333/sorc/gdas.cd/build$ ctest -R test_gdasapp_atm_jjob_var_run
Test project /scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/pr333/sorc/gdas.cd/build
    Start 1352: test_gdasapp_atm_jjob_var_run
1/1 Test #1352: test_gdasapp_atm_jjob_var_run ....   Passed  138.08 sec

100% tests passed, 0 tests failed out of 1

Total Test time (real) = 138.39 sec
(gdasapp) Hera(hfe04):/scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/pr333/sorc/gdas.cd/build$ ctest -R test_gdasapp_atm_jjob_ens_run
Test project /scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/pr333/sorc/gdas.cd/build
    Start 1355: test_gdasapp_atm_jjob_ens_run
1/1 Test #1355: test_gdasapp_atm_jjob_ens_run ....   Passed  298.36 sec

100% tests passed, 0 tests failed out of 1

Total Test time (real) = 298.67 sec
```

This confirms that the original failure of these tests on Hera is due to the observation files.

CoryMartin-NOAA commented:
@RussTreadon-NOAA yes sorry I didn't get a chance to start the sync yesterday once Hera was back up for use. Will sync files from Orion to Hera this morning.

RussTreadon-NOAA commented:
> @RussTreadon-NOAA yes sorry I didn't get a chance to start the sync yesterday once Hera was back up for use. Will sync files from Orion to Hera this morning.

Perfect!  Thank you @CoryMartin-NOAA

RussTreadon-NOAA commented:
Given @ADCollard's approval and @CoryMartin-NOAA'sprevious approval, I'll merge this PR into develop.
